src/NHibernate/Tuple/Entity/AbstractEntityTuplizer.cs

```diff
@@ -114,6 +114,11 @@ public object Instantiate(object id)
 
 		public object GetIdentifier(object entity)
 		{
+			if (entity is INHibernateProxy proxy)
```

Comment:
In other places the caller checks for proxy before calling this method.

Reply:
Indeed, in `NHibernate.Type.EntityType`, the checks done there are no more needed with this PR changes.

There are other cases where the proxy is unproxied before hand, but that looks to me as a requirement of the implemented features there than due to `GetIdentifier` not supporting proxies.

Reply:
> In other places the caller checks for proxy before calling this method.

And your point? You prefer to keep it that way? To me it looks better to support it here.
Or at least move this check to `AbstractEntityPersister.GetIdentifier`.

If it's about clean up other places - I think we can't remove checks in other places in this PR to avoid some custom implementations relying on it. I can add 6.0 TODO here for clean up.

Reply:
I would prefer to keep this check outside of this method as in other places.

Reply:
So how about moving it to `AbstractEntityPersister.GetIdentifier`?  As  I don't see why `persister.GetIdentifier` shouldn't handle proxies.

Also I've just noticed the following code:
https://github.com/nhibernate/nhibernate-core/blob/8d13f2492ecd30a1fc451eee38208dbf95a2a614/src/NHibernate/Engine/ForeignKeys.cs#L156-L159
I think we need to use the same logic in both places. But not sure which.

Reply:
I am seeing more benefits in having the checks inside that method rather than outside. I do not really see why this method should not do the check itself. It does look very lightweight to me, so having the check done for cases where it can never be a proxy should not have a significant impact. What am I missing?

Reply:
@fredericDelaporte I agree with you. But if you are OK with my latest changes I think this part is out of scope of this PR as now IsTransient logic never calls `GetIdentifier` for proxies